fpt/src/debugger.rs

```diff
@@ -0,0 +1,54 @@
+use crate::debug_interface::DebugCmd;
+
+#[derive(Clone, PartialEq)]
+pub struct Breakpoint {
+    pub pc: u16,
+    pub active: bool,
+}
+#[derive(Clone, PartialEq)]
+pub struct Watchpoint {
+    pub addr: u16,
+}
+
+#[derive(Clone, PartialEq)]
+pub struct Debugger {
+    breakpoints: Vec<Breakpoint>,
+    watchpoints: Vec<Watchpoint>,
+    pub paused: bool,
+}
+
+impl Debugger {
+    pub fn new() -> Self {
+        Self {
+            breakpoints: Vec::new(),
+            watchpoints: Vec::new(),
+            paused: false,
+        }
+    }
+
+    pub fn receive_command(&mut self, cmd: &DebugCmd) {
```

Comment:
This signature matches the one in `trait DebugInterface`.
Why not put this in a `impl DebugInterface for Debugger`?